block/retriever.go

```diff
@@ -135,8 +137,12 @@ func (m *Manager) ProcessNextDABatch(daMetaData *da.DASubmitMetaData) error {
 				return fmt.Errorf("apply block: height: %d: %w", block.Header.Height, err)
 			}
 
-			delete(m.blockCache, block.Header.Height)
+			lastAppliedHeight = float64(block.Header.Height)
+
+			m.blockCache.DeleteBlockFromCache(block.Header.Height)
 		}
+
+		types.LastReceivedDAHeightGauge.Set(lastAppliedHeight)
```

Comment:
Why not just 1 liner 
types.LastReceivedDAHeightGauge.Set( float64(block.Header.Height))

Reply:
no need to set it every time a block is applied